coap/utils.go

```diff
@@ -0,0 +1,120 @@
+package coap
+
+import (
+	"encoding/binary"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	gocoap "github.com/dustin/go-coap"
+	"github.com/gogo/protobuf/proto"
+	"github.com/mainflux/mainflux"
+	broker "github.com/nats-io/go-nats"
+)
+
+// AuthProvider represents ManagerClient which provides auth features.
+type AuthProvider interface {
```

Comment:
Please note that some of the functions below can be bound to the `AuthProvider` instead of passing the provider to them.